ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/manager/AbstractEthTask.java

```diff
@@ -117,6 +127,17 @@ public final T result() {
   /** Execute core task logic. */
   protected abstract void executeTask();
 
+  /** Executes the task while timed by a timer. */
+  public void executeTaskTimed() {
+    final OperationTimer.TimingContext timingContext = taskTimer.startTimer();
+    executeTask();
+    taskTimeInSec = timingContext.stopTimer();
```

Comment:
We should probably use the `try (final OperationTimer.TimingContext timingContext = taskTimer.startTimer())` or an explicit `final` block here to ensure the timer completes even if the task fails.

Reply:
We need the final block because close doesn't return a time and we want to keep that for logging at higher levels.